### PR TITLE
Go example plugin

### PR DESCRIPTION
**Pull request by qlikmats**
Fixes #[issue number].

### Status
```
[x] Under development
[x] Waiting for code review
[ ] Waiting for merge
```
### Information
```
[ ] Contains breaking changes
[ ] Contains new API(s)
[x] Contains documentation
[ ] Contains test
[ ] Contains examples
```
### To-do list
```
[ ] 
[ ] 
[ ] ...
```

### CODE REVIEW

File: examples/go/basic_example/server.go
Comment (carlioth) [inline examples/go/basic_example/server.go]: This is an issue when running on mac, change to //go:generate protoc -I ../../../proto ../../../proto/ServerSideExtension.proto --go_out=plugins=grpc:./gen
Guess that should work on Windows, please verify

File: examples/go/basic_example/server.go
Comment (carlioth) [inline examples/go/basic_example/server.go:256]: Add some initial information logging, what port it's running on.

**Review by carlioth — COMMENTED** [review]
Nice example. 
I've been trying it out to run containerized, some thoughts from me in comments. 
It would also be nice to add an qvf, consistent with the other examples.

File: examples/go/basic_example/server.go
Comment (qlikmats) [reply examples/go/basic_example/server.go]: Indeed it works, and I have changed to forward-slash. Thanks.

File: examples/go/basic_example/server.go
Comment (qlikmats) [reply examples/go/basic_example/server.go:256]: Good point. Will add that.

**Comment by carlioth** [comment]
Another thing to consider is to use dep, as dependency management. Also consider checking in you vendor (dependencies). You can read more / see example here: https://github.com/qlik-trial/cloud-app-retriever#dependency-management
More about dep here: https://hackernoon.com/using-go-dep-as-a-project-maintainer-641d1f3006d7

File: examples/go/README.md
Comment (qlik-aoh) [inline examples/go/README.md]: Remove blank.

File: examples/go/README.md
Comment (qlik-aoh) [inline examples/go/README.md]: More of a script than a go command.
Also, we should aim for platform independence and the script is windows only.

File: examples/go/README.md
Comment (qlik-aoh) [inline examples/go/README.md]: That's not a very helpful description. Howabout

The GetCapabilities method returns a [Capabilities](https://github.com/qlik-oss/server-side-extension/blob/master/docs/SSE_Protocol.md#qlik.sse.Capabilities) object describing the operations supported by the plugin.

File: examples/go/basic_example/server.go
Comment (qlik-aoh) [inline examples/go/basic_example/server.go]: Shouldn't the path be relative rather than point out your fork repo?

**Review by qlik-aoh — CHANGES_REQUESTED** [review]
Perhaps you could some CircleCi test using this plugin?

File: examples/go/README.md
Comment (qlikmats) [reply examples/go/README.md]: I was thinking of having Windows/Linux scripts, but I can scrap the scripts altogether and just document the command line(s).

File: examples/go/README.md
Comment (qlikmats) [reply examples/go/README.md]: copy/paste "error" from one of the other examples, so I will change here and make a comment for the other plugin as well :)

File: examples/go/basic_example/server.go
Comment (qlikmats) [reply examples/go/basic_example/server.go]: Don't think it should be relative, if I have understood Go import paths correctly, but it should not point to my fork when "merging" to the parent repo. For now, I think it needs to be like this in order to work...

**Review by qlik-aoh — APPROVED** [review]

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: This link has to be changed before merging to qlik-oss.

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: This link has to be changed before merging to qlik-oss.

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: Make the steps in here to a list of points instead. Easier to follow that way.

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: "Then" should be the start of a new point/step.

Comment (tobiaslindulf) [inline examples/go/README.md]: You can also link to the https://github.com/golang/protobuf#installation page where this is described in a good way.

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: I had problems to find the protoc-gen-go locally on windows after this command. Finally I discovered it ended up in C:\Users\\[User]\go\bin .

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: It was not obvious to me, which zip to choose from in the list. I had to scroll down quite long on that page to understand that I had to choose the windows zip.
Maybe state something like "download one of the pre-built binaries in zip packages: protoc-$VERSION-$PLATFORM.zip".

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: Without changing the code, port 50055 was opened, not 50051.

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: Without changing the code, port 50055 was opened, not 50051.

Comment (tobiaslindulf) [inline examples/go/README.md]: The qvf app refers to SSEGo, not SSE_Go.

File: examples/go/basic_example/server.go
Comment (tobiaslindulf) [inline examples/go/basic_example/server.go]: Add a logline saying log.Printf("GetCapabilities called")

File: examples/go/README.md
Comment (tobiaslindulf) [inline examples/go/README.md]: To be consistent between our different examples I think we should add some more examples:
-SumOfCol: To have an aggregated example.
-Cache and NoCache functions to show how to manipulate headers.

**Review by tobiaslindulf — CHANGES_REQUESTED** [review]
Also add a Go section in https://github.com/qlikmats/server-side-extension/blob/master/README.md . Just like we have links to the other languages.
Consider updating the repo from qlik-oss master first to get the java changes in before merging (perhaps not needed).

File: examples/go/README.md
Comment (qlikmats) [reply examples/go/README.md]: Yes, fully aware of that.

File: examples/go/README.md
Comment (qlikmats) [reply examples/go/README.md]: Yes, fully aware of that.

File: README.md
Comment (qlikmats) [inline README.md]: Add this README.md

File: examples/go/README.md
Comment (qlikmats) [reply examples/go/README.md]: Done.

File: examples/go/basic_example/server.go
Comment (qlikmats) [reply examples/go/basic_example/server.go]: Done.

File: examples/go/basic_example/README.md
Comment (josefinestal) [inline examples/go/basic_example/README.md]: This path does not seem to work. I think you might need another ../

File: examples/go/basic_example/README.md
Comment (josefinestal) [inline examples/go/basic_example/README.md]: The dash before the "[" disappears when viewing the file. I think it's enough to add another "\" to solve it.

**Review by josefinestal — APPROVED** [review]
I mainly focused on the documentation and trying to run the example. A part from the comments I gave it looks good, and the example runs smoothly.

File: examples/go/README.md
Comment (qlikmats) [inline examples/go/README.md]: "There are five functions..."

**Review by tobiaslindulf — APPROVED** [review]

**Pull request merged**
